### PR TITLE
Closes #137 LRC visual regression on live templates

### LINKED ISSUE #137: LRC visual regression on live templates
Scenario: Shouldn't cause visual regression when visiting x template
Given: 3.17+ is active
And: Visit x page
When: clear cache 
And: visit the page
then: no visual regression compared to nowprocket

Templates:
delayjs_1.3_live_template.php 
delayjs_68_live_template.php
page lcp_6647_live.php
lcp_live_4_template.php
lcp_live_test_span.php
rucss_5213_template.php
template 4787.php
lrc_will_site.php

### PR DESCRIPTION
**Pull request by Khadreal**
# Description

Fixes #137 

## Type of change

- [x] New feature (non-breaking change which adds functionality).

## Detailed scenario
Allow QA to test for any visual regression on LRC templates

### Documentation
Allow QA to test for any visual regression on LRC templates

### New dependencies
N/A

### Risks
N/A

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style
- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

### CODE REVIEW

**Review by Miraeld — DISMISSED** [review]

File: src/features/lrc-visual-regression.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  Background:
+    Given I am logged in
+    And plugin is installed 'new_release'
+    And plugin is activated
+    And I go to 'wp-admin/options-general.php?page=wprocket#dashboard'
+
+    Scenario: Shouldn't have any visual regression when visiting
+      When I log out
+      And I visit lrc live templates
```
Comment (jeawhanlee) [inline src/features/lrc-visual-regression.feature]: Can we make this test generic, some more like `And I visit scenario urls`

File: src/support/steps/general.ts [hunk 2]
```diff
@@ -245,7 +245,8 @@ When('I clear cache', async function (this:ICustomWorld) {
     await this.utils.gotoWpr();
 
     this.sections.set('dashboard');
-    await this.sections.toggle('clearCacheBtn');
+    await this.page.locator("text=Clear and preload").last().click();
```
Comment (jeawhanlee) [inline src/support/steps/general.ts]: Does this work as expected even when preload is disabled?

File: src/support/steps/general.ts [hunk 3]
```diff
@@ -301,6 +317,17 @@ Then('I must not see any visual regression {string}', async function (this: ICus
...
 });
 
+/**
+ * Executes the step to check for LRC visual regression.
+ */
+Then('I must not see any visual regression in scenario urls', async function (this: ICustomWorld) {
+    const liveUrl = SCENARIO_URLS;
+
+    for (const key in liveUrl) {
+        await compareReference(liveUrl[key]);
```
Comment (jeawhanlee) [inline src/support/steps/general.ts]: Should this not be the key instead?

**Review by jeawhanlee — CHANGES_REQUESTED** [review]

**Comment by jeawhanlee** [comment]
Currently backstop tests are not running well, because of my comment [here](https://github.com/wp-media/wp-rocket-e2e/pull/145#discussion_r1772770188)

File: src/support/steps/general.ts (hunk 2 above)
Comment (Khadreal) [reply src/support/steps/general.ts]: Didn't test with preload disabled, will check

Comment (jeawhanlee) [reply src/support/steps/general.ts]: You can use the `pre_get_rocket_option_manual_preload` filter

Comment (jeawhanlee) [reply src/support/steps/general.ts]: I can confirm this is fixed now.

**Review by jeawhanlee — DISMISSED** [review]

**Review by jeawhanlee — DISMISSED** [review]

File: config/scenarioUrls.json [hunk 4]
```diff
@@ -20,40 +20,66 @@
...
+      "path": "delayjs_68_live_template"
+    },
+    "lcp6647Live": {
+      "path": "lcp_6647_live"
+    },
+    "lcpLive4Template": {
+      "path": "lcp_live_4_template"
+    },
+    "lcpLiveTestSpan": {
+      "path": "lcp_live_test_span"
```
Comment (Mai-Saad) [inline config/scenarioUrls.json:37]: Let's remove this template from e2e as it's failing cause of edge acceptable case https://wp-media.slack.com/archives/CUKB44GNN/p1727454310142779?thread_ts=1727444219.707339&cid=CUKB44GNN

**Comment by Mai-Saad** [comment]
@Khadreal Thanks for the PR 
- test itself is passing
- backstop for VR  (4passes/8) => Note for lcp_live_4 , removed recaptcha js so that it works {pushed to test repo}
- Now failed templates that may need update or removal:
delayjs_1-3_live_template
lcp_live_test_span
rucss_5213_template
lrcWillSite

Screenshot from 2024-10-04 13-35-33
Screenshot from 2024-10-04 13-35-28
Screenshot from 2024-10-04 13-35-24
Screenshot from 2024-10-04 13-35-20
Screenshot from 2024-10-04 13-35-16
Screenshot from 2024-10-04 13-35-12
Screenshot from 2024-10-04 13-35-08
Screenshot from 2024-10-04 13-35-04
Screenshot from 2024-10-04 13-33-28

**Comment by Mai-Saad** [comment]
- If WPR was installed before running the test, it won't be removed when running the tests causing failure at install step (further investigations needed)
- Failed results analysis on latest backup of e2e https://wp-media.slack.com/archives/CUT7FLHF1/p1728555417016799?thread_ts=1728454435.427329&cid=CUT7FLHF1

**Comment by Khadreal** [comment]
I did another round of test and the plugin was removed when the test was running. @jeawhanlee could help run the test when you have some time.

**Comment by jeawhanlee** [comment]
I can confirm with @Khadreal that the plugin will be removed from the env if it was already installed before running tests.

**Comment by Mai-Saad** [comment]
opened another ticket for the uninstall issue https://github.com/wp-media/wp-rocket-e2e/issues/153

**Review by Mai-Saad — APPROVED** [review]
Handle failed templated can be on other PR https://github.com/wp-media/wp-rocket-e2e/issues/154

**Pull request merged**